mewarpx/mewarpx/mwxrun.py

```diff
@@ -135,12 +165,150 @@ def init_grid(self, lower_bound, upper_bound, number_of_cells, **kwargs):
+
+        if isinstance(self.grid, picmi.Cartesian1DGrid):
+            self.xmin = 0.0
+            self.xmax = 1.0
+            self.ymin = 0.0
+            self.ymax = 1.0
+            self.zmin = self.grid.xmin
+            self.zmax = self.grid.xmax
+            self.rmin = None
+            self.rmax = None
+
+            self.nx = 0
+            self.ny = 0
+            self.nz = self.grid.nx
+            self.nr = None
+
+            self.dx = None
+            self.dy = None
+            self.dz = (self.zmax - self.zmin) / self.nz
+            self.dr = None
+
+        elif isinstance(self.grid, picmi.Cartesian2DGrid):
+            self.xmin = self.grid.xmin
+            self.xmax = self.grid.xmax
+            self.ymin = 0.0
+            self.ymax = 1.0
+            self.zmin = self.grid.ymin
+            self.zmax = self.grid.ymax
+            self.rmin = None
+            self.rmax = None
+
+            self.nx = self.grid.nx
+            self.ny = 0
+            self.nz = self.grid.ny
+            self.nr = None
+
+            self.dx = (self.xmax - self.xmin) / self.nx
+            self.dy = None
+            self.dz = (self.zmax - self.zmin) / self.nz
+            self.dr = None
+
+        elif isinstance(self.grid, picmi.Cartesian3DGrid):
+            self.xmin = self.grid.xmin
+            self.xmax = self.grid.xmax
+            self.ymin = self.grid.ymin
+            self.ymax = self.grid.ymax
+            self.zmin = self.grid.zmin
+            self.zmax = self.grid.zmax
+            self.rmin = None
+            self.rmax = None
+
+            self.nx = self.grid.nx
+            self.ny = self.grid.ny
+            self.nz = self.grid.ny
+            self.nr = None
+
+            self.dx = (self.xmax - self.xmin) / self.nx
+            self.dy = (self.ymax - self.ymin) / self.ny
+            self.dz = (self.zmax - self.zmin) / self.nz
+            self.dr = None
+
+        elif isinstance(self.grid, picmi.CylindricalGrid):
+            self.xmin = self.ymin = self.rmin = self.grid.rmin
+            self.xmax = self.ymax = self.rmax = self.grid.rmax
+            self.zmin = self.grid.zmin
+            self.zmax = self.grid.zmax
+
+            self.nx = self.ny = self.nr = self.grid.nr
+            self.nz = self.grid.nz
+
+            self.dx = self.dy = self.dr = (self.rmax - self.rmin) / self.nr
+            self.dz = (self.zmax - self.zmin) / self.nz
+
+        else:
+            raise ValueError("Unrecognized type of pywarpx.picmi Grid.")
+
+    def _print_grid_params(self):
+        """Function to print out the grid parameters."""
+        if self.geom_str == 'RZ':
+            logger.info(
+                f"Creating grid with nr={self.nr}, nz={self.nz} "
+                f"and r, z limits of [[{self.rmin:.4g}, {self.rmax:.4g}], "
+                f"[{self.zmin:.4g}, {self.zmax:.4g}]]"
+            )
+        else:
+            logger.info(
+                f"Creating grid with nx={self.nx}, ny={self.ny}, nz={self.nz} "
```

Comment:
```suggestion
                f"Creating {self.geom_str} grid with nx={self.nx}, ny={self.ny}, nz={self.nz} "
```

Reply:
Sorry, you'll have to redo line-breaks if you accept this, but explicitly specifying the dimension seems helpful.